src/shared/modules.mjs

```diff
@@ -854,6 +854,167 @@ const generateEventListenResponse = json => {
+                anyOf.schema.allOf = anyOf.allOf
+                delete anyOf.allOf
+            }
+        }
+    }
+    anyOfType.name = name[0].toLowerCase() + name.substr(1)
+    anyOfType.summary = summary
+    anyOfType.schema = anyOf
+    return anyOfType
+}
+
+const generateParamsAnyOfSchema = (methodParams, anyOf, anyOfTypes, title, summary) => {
+    let params = []
+    methodParams.forEach(p => {
+        if (p.schema === anyOfTypes) {
+            let anyOfType = generateAnyOfSchema(anyOf, title, summary)
+            anyOfType.required = p.required
+            params.push(anyOfType)
+        }
+        else {
+            params.push(p)
+        }
+    })
+    return params
+}
+
+const generateResultAnyOfSchema = (method, methodResult, anyOf, anyOfTypes, title, summary) => {
+    let methodResultSchema = Object.assign({}, method.result.schema)
+
+    if (methodResult.schema === anyOfTypes) {
+        let anyOfType = generateAnyOfSchema(anyOf, title, summary)
+        let index = 0
+        if (isEventMethod(method)) {
+            index = (method.result.schema.anyOf || method.result.schema.oneOf).indexOf(getPayloadFromEvent(method))
+        }
+        else {
+            index = (method.result.schema.anyOf || method.result.schema.oneOf).indexOf(anyOfType)
+        }
+        if (method.result.schema.anyOf) {
+            methodResultSchema.anyOf = Object.assign([], method.result.schema.anyOf)
+            methodResultSchema.anyOf[index] = anyOfType.schema
+        }
+        else if (method.result.schema.oneOf) {
+            methodResultSchema.oneyOf = Object.assign([], method.result.schema.oneOf)
+            methodResultSchema.oneOf[index] = anyOfType.schema
+        }
+        else {
+            methodResultSchema = anyOfType.schema
+        }
+    }
+    return methodResultSchema
+}
+
+const createPolymorphicMethods = (method, json) => {
+    let anyOfTypes = []
+    let methodParams = []
+    let methodResult = Object.assign({}, method.result)
+
+    method.params.forEach(p => {
+        if (p.schema) {
+            methodParams.push(getAnyOfSchema(p, json, anyOfTypes))
+        }
+    })
+    if (isEventMethod(method)) {
+        methodResult.schema = getPayloadFromEvent(method)
+    }
+    methodResult = getAnyOfSchema(methodResult, json, anyOfTypes)
+
+    let polymorphicMethodSchemas = []
+    if (anyOfTypes.length > 0) {
+        let polymorphicMethodSchema = {
+            name: {},
+            tags: {},
+            summary: `${method.summary}`,
+            params: {},
+            result: {},
+            examples: {}
+        }
+        anyOfTypes.forEach(anyOf => {
+            let localized = localizeDependencies(anyOf, json)
+            let title = localized.title || localized.name || ''
+            let summary = localized.summary || localized.description || ''
+            polymorphicMethodSchema.actualName = method.name
+            polymorphicMethodSchema.name = `${method.name}With${title}`
```

Comment:
I was assuming that if the params have an anyOf, we'd use <method.name>With<title> for the new method, but if the result had an anyOf , we'd use `<method.name><Title>` since method names are generally verbs, you can say "navigateToHomeIntent" since HomeIntent is a result, but if you had a method called `walk(p: pet)` (and pet can be cat or dog) you would name them `walkWithDog` and `walkWithCat`

Reply:
Is there  a way in javascript to identify the string based on the value ?

Reply:
done